corehq/util/test_utils.py

```diff
@@ -169,6 +170,40 @@ class flag_disabled(flag_enabled):
     enabled = False
 
 
+@contextmanager
+def disable_overpatch(patch_obj):
+    """Prevent class decorator from masking method decorator"""
+    def apply_to(patch):
+        def decorate_callable(func):
+            key = getkey(patch)
+            if any(key == getkey(p) for p in getattr(func, "patchings", [])):
+                # do not decorate if already patched for this oneshot
+                return func
+            return real_decorate_callable(func)
+
+        def copy():
+            return apply_to(real_copy())
+
+        real_decorate_callable = patch.decorate_callable
+        real_copy = patch.copy
+        patch.decorate_callable = decorate_callable
+        patch.copy = copy
+        patches.append(patch)
+        return patch
+
+    def getkey(patch):
+        return patch.getter(), patch.attribute
+
+    patches = []
```

Comment:
would you be open to moving this above the `apply_to` definition since that manipulates its state?  Feels weird to have something declared after it's used - I actually didn't realize this would be valid

Reply:
`apply_to` is not called before `patches` is bound to its value. I'd prefer to keep it here; it was an intentional decision to define the functions before the code that calls them.